torch/nn/modules/_functions.py

```diff
@@ -16,17 +17,25 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
         if size == 1 and world_size < 2:
             raise ValueError('Expected more than 1 value per channel when training, got input size {}'.format(size))
 
-        # calculate mean/invstd for input.
-        mean, invstd = torch.batch_norm_stats(input, eps)
+        num_channels = input.shape[1]
+        if input.numel() > 0:
+            # calculate mean/invstd for input.
+            mean, invstd = torch.batch_norm_stats(input, eps)
 
-        count = torch.full((1,), input.numel() // input.size(1),
-                           dtype=mean.dtype,
-                           device=mean.device)
+            count = torch.full((1,), input.numel() // input.size(1),
+                            dtype=mean.dtype,
+                            device=mean.device)
 
+            # C, C, 1 -> (2C + 1)
+            combined = torch.cat([mean, invstd, count], dim=0)
+        else:
+            # for empty input, directly set all stats to 0
```

Comment:
I'm not sure this comment is accurate. The code works because all inputs for which the count is 0 are filtered out later. Not because the stats contain 0s.

Reply:
added more detailed descriptions.